src/HydraAuction/Delegate.hs

```diff
@@ -57,37 +54,56 @@ execDelegateRunnerT (MkDelegateRunner action) = evalStateT action initialState
 instance MonadTrans DelegateRunnerT where
   lift = MkDelegateRunner . lift
 
-delegateStep ::
+type ClientId = Int
+
+data ClientResponseScope
+  = Broadcast
+  | PerClient ClientId
+
+clientIsInScope :: ClientId -> ClientResponseScope -> Bool
+clientIsInScope clientId scope = case scope of
+  Broadcast -> True
+  PerClient expectedClientId -> clientId == expectedClientId
+
+delegateFrontendRequestStep ::
   forall m.
   Monad m =>
-  DelegateInput ->
-  DelegateRunnerT m [DelegateResponse]
-delegateStep input = case input of
-  -- FIXME: initialize Hydra
-  DelegateEvent Start -> return []
-  -- FIXME: close Hydra node
-  DelegateEvent (AuctionStageStarted BiddingEndedStage) -> return []
-  -- FIXME: abort Hydra node
-  DelegateEvent (AuctionStageStarted VoucherExpiredStage) -> return []
-  DelegateEvent (AuctionStageStarted _) -> return []
-  -- FIXME: commit empty transaction
-  DelegateEvent (HydraEvent (Committed _)) -> return []
-  -- FIXME: fanout Hydra node
-  DelegateEvent (HydraEvent ReadyToFanout) -> return []
-  DelegateEvent (HydraEvent _) -> return []
+  (ClientId, FrontendRequest) ->
+  DelegateRunnerT m [(ClientResponseScope, DelegateResponse)]
```

Comment:
In case of returning multiple responses. Like one for broadcast and one per-request. It is not used now, but maybe there is some case where this is required. Probably we do not need it, but it does not seem to hurt now, no?

Reply:
I mean it’s a bit a distinction between a list of pairs and a pair of lists. (Besides perhaps performance characteristics)

Reply:
Pair of lists does not seem to work here, cuz we need to process in pairs here.

Also we may need change this pair into ADT later, cuz `delegateXStep` may do other kinds of responses, like adding DelegateEvent.

Reply:
@uhbif19 I think you are right. As a matter of fact, in your next PR I see a good case for returning multiple responses.

When a client places a `newBid` we want to: confirm the bid is placed and give the user the ClosingTxTemplate, but also broadcast an update of the delegate state.